composite/market-closure/src/config.ts

```diff
@@ -0,0 +1,21 @@
+import { getDefaultConfig } from '@chainlink/external-adapter'
+import { util } from '@chainlink/ea-bootstrap'
+import { getDataProvider, PriceAdapter } from './dataProvider'
+import { Check, getCheckImpl, getCheckProvider } from './checks'
+
+export type Config = {
+  priceAdapter: PriceAdapter
+  checkAdapter: Check
+}
+
+export const makeConfig = (prefix = ''): Config => {
+  const dataProviderUrl = util.getRequiredEnv('DATA_PROVIDER_URL', prefix)
```

Comment:
So doing it this way feeds would still have to run one market-closure adapter for finnhub and one market-closure adapter for fcsapi right?

What if instead we made this more flexible. It could take in a new param `provider` of either `finnhub` or `fcsapi` that we use as the prefix for the `DATA_PROVIDER_URL`. The `baseURL` is switched depending on what provider you want.

We could do the same with `check`.

I used this pattern in mine, https://github.com/smartcontractkit/external-adapters-js/pull/349/files#diff-3312b3574a4a6a6050bc295ea85bc593cdde939e8abf11a51df6a46151bfb65dR11. See what you think.

This would let feeds just run one single `market-closure` adapter.

Reply:
We unfortunately can't split this on input params, as the CL node will send the exact same input params to each adapter. Going to leave a comment on your other PR as well.